inc/class-pressbookssentry.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * @author  Pressbooks <code@pressbooks.com>
+ * @license GPLv3 (or any later version)
+ */
+
+namespace Pressbooks;
+
+use function \Pressbooks\Utility\debug_error_log;
+use PressbooksMix\Assets;
+
+class PressbooksSentry {
+
+	const DEFAULT_ENVIRNOMENT = 'staging';
+
+	const DEFAULT_TRACE_SAMPLE_RATE = 0.5;
+
+	const WP_SCRIPT_NAME = 'script-sentry';
+
+	const ENABLED_VALUE = 1;
+
+	const ENVIRONMENT_VARIABLES_V1 = 1;
+
+	const ENVIRONMENT_VARIABLES_V2 = 2;
+
+	/**
+	 * @var PressbooksSentry
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Sentry DSN
+	 *
+	 * @var string\bool
+	 */
+	private $dsn = false;
+
+	/**
+	 * @var \WP_User
+	 */
+	private $user;
+
+	/**
+	 * @since 5.5.3
+	 *
+	 * @return PressbooksSentry
+	 */
+	static public function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+			self::$instance->setSentryDSNFromEnvironmentVariables();
+			self::$instance->getCurrentUserForTracking();
+			if (
+				self::environmentVariablesType() === self::ENVIRONMENT_VARIABLES_V1 ||
```

Comment:
If [previous integration version](https://github.com/pressbooks/private/issues/130#issuecomment-634938654) 
```
SENTRY_KEY=
SENTRY_ORGANIZATION=
SENTRY_PROJECT=
```
is implemented, only PHP issues will be captured by Sentry.